aws/resource_aws_dms_endpoint.go

```diff
@@ -333,13 +409,41 @@ func resourceAwsDmsEndpointSetState(d *schema.ResourceData, endpoint *dms.Endpoi
 	d.Set("endpoint_type", strings.ToLower(*endpoint.EndpointType))
 	d.Set("engine_name", endpoint.EngineName)
 
-	if *endpoint.EngineName == "dynamodb" {
+	switch *endpoint.EngineName {
+	case "dynamodb":
 		if endpoint.DynamoDbSettings != nil {
 			d.Set("service_access_role", endpoint.DynamoDbSettings.ServiceAccessRoleArn)
 		} else {
 			d.Set("service_access_role", "")
 		}
-	} else {
+	case "mongodb":
+		if endpoint.MongoDbSettings != nil {
+			d.Set("username", endpoint.MongoDbSettings.Username)
+			d.Set("server_name", endpoint.MongoDbSettings.ServerName)
+			d.Set("port", endpoint.MongoDbSettings.Port)
+			d.Set("database_name", endpoint.MongoDbSettings.DatabaseName)
+
+			d.Set("mongodb_settings.0.auth_type", endpoint.MongoDbSettings.AuthType)
+			d.Set("mongodb_settings.0.auth_mechanism", endpoint.MongoDbSettings.AuthMechanism)
+			d.Set("mongodb_settings.0.nesting_level", endpoint.MongoDbSettings.NestingLevel)
+			d.Set("mongodb_settings.0.extract_doc_id", endpoint.MongoDbSettings.ExtractDocId)
+			d.Set("mongodb_settings.0.docs_to_investigate", endpoint.MongoDbSettings.DocsToInvestigate)
+			d.Set("mongodb_settings.0.auth_source", endpoint.MongoDbSettings.AuthSource)
+		} else {
+			d.Set("username", endpoint.Username)
+			d.Set("server_name", endpoint.ServerName)
+			d.Set("port", endpoint.Port)
+			d.Set("database_name", endpoint.DatabaseName)
+
+			// Default values as per https://docs.aws.amazon.com/dms/latest/userguide/CHAP_Source.MongoDB.html
```

Comment:
We should move these to the `Default` parameter in the resource schema: https://www.terraform.io/docs/extend/schemas/schema-behaviors.html#default

e.g. 
```go
"auth_type": {
  Type:     schema.TypeString,
  Optional: true,
  Default: "PASSWORD",
},
```